homeassistant/components/sql/util.py

```diff
@@ -223,3 +226,16 @@ def generate_lambda_stmt(query: str) -> StatementLambdaElement:
     """Generate the lambda statement."""
     text = sqlalchemy.text(query)
     return lambda_stmt(lambda: text, lambda_cache=_SQL_LAMBDA_CACHE)
+
+
+def convert_value(value: Any) -> Any:
+    """Convert value."""
+    match value:
+        case Decimal():
+            return float(value)
+        case date() | datetime():
```

Comment:
```suggestion
        case date():
```
We should not convert `datetime`, it will be done by the base entity